xls/dslx/lsp/language_server_adapter.cc

```diff
@@ -82,14 +83,54 @@ void AppendDiagnosticFromTypecheck(
   }
 }
 
+// If the span's filename is a relative path, we map it into a URI by finding
+// its path on disk.
+//
+// Note: there may be a more holistic way to do this long term -- the LSP
+// effectively requires us to create a filesystem overlay that mixes real files
+// and in-memory files with modifications. For now, this at least enables
+// things like go-to-definition in files that have not yet been loaded into the
+// language server.
+absl::StatusOr<std::string> MaybeRelpathToUri(
```

Comment:
There are also some path/uri functions in [`verible/common/lsp/lsp-file-utils.h`](https://github.com/chipsalliance/verible/blob/master/common/lsp/lsp-file-utils.h), though probably not exactly what you need here. Mostly an FYI.

Reply:
Thanks, good to know -- if I see more actually-URI-looking-encoded stuff those look helpful.